azure/services/scalesets/scalesets.go

```diff
@@ -372,6 +372,28 @@ func (s *Service) validateSpec(ctx context.Context) error {
 		}
 	}
 
+	// Validate DiagnosticProfile spec
+	if spec.DiagnosticsProfile.Boot != nil {
+		if spec.DiagnosticsProfile.Boot.StorageAccountType == infrav1.UserManagedDiagnosticsStorage {
+			if spec.DiagnosticsProfile.Boot.UserManaged == nil {
+				return azure.WithTerminalError(fmt.Errorf("userManaged must be specified when storageAccountType is '%s'", infrav1.UserManagedDiagnosticsStorage))
+			} else if spec.DiagnosticsProfile.Boot.UserManaged.StorageAccountURI == "" {
+				return azure.WithTerminalError(fmt.Errorf("storageURI cannot be empty when storageAccountType is '%s'", infrav1.UserManagedDiagnosticsStorage))
+			}
+		}
```

Comment:
I think @CecileRobertMichon is referring to remove this block of validation for scale sets in [the comment above.](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2528/files#r1043876133)